dotCMS/src/main/java/com/liferay/portlet/PortletURLImpl.java

```diff
@@ -193,11 +193,17 @@ public void setEncrypt(boolean encrypt) {
 	}
 
 	public String toString() {
-		StringBuffer sb = new StringBuffer();
 
-		sb.append(PortalUtil.getPortalURL(_req, _secure));
+        String ctxPath = (_portletReq != null) ?
+            ctxPath = (String)_portletReq.getPortletSession().getAttribute(

```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract the assignment out of this expression. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AAssignmentInSubExpressionCheck)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "ctxPath". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1854)